spec/javascripts/services/gis_service_spec.coffee

```diff
@@ -0,0 +1,131 @@
+do ->
+  'use strict'
+  describe 'GISService', ->
+    GISService = undefined
+    httpBackend = undefined
+    $q = undefined
+    $rootScope = undefined
+    fakeShortFormDataService =
+      formatUserDOB: ->
+    fakeValidAddressMatchResult = getJSONFixture('gis-api-valid-address-boundary-match.json')
+    fakeValidAddressNoMatchResult = getJSONFixture('gis-api-valid-address-no-boundary-match.json')
+    fakeInvalidResult = getJSONFixture('gis-api-invalid-address.json')
+    requestURL = undefined
+    fakeListing = getJSONFixture('listings-api-show.json')
+    fakeAddress =
+      address1: '4053 18th St'
+      city: 'San Francisco'
+      state: 'CA'
+      zip: '94114'
+    fakeApplicant =
+      firstName: 'Bob'
+      lastName: 'Williams'
+      dob_month: '07'
+      dob_day: '05'
+      dob_year: '2015'
+      preferences: {liveInSf: null, workInSf: null}
+    fakeHouseholdMember =
+      firstName: 'Ethel'
+      lastName: 'O\'Keefe'
+      dob_month: '09'
+      dob_day: '27'
+      dob_year: '1966'
+    gisDataOptions =
+      address: fakeAddress
+      listing: fakeListing
+      member: fakeHouseholdMember
+      applicant: fakeApplicant
+      project_id: fakeListing.Project_ID
+
+    beforeEach module('ui.router')
+    beforeEach module('dahlia.services', ($provide)->
+      $provide.value 'ShortFormDataService', fakeShortFormDataService
+      return
+    )
+
+    beforeEach inject((_GISService_, _$httpBackend_, _$q_, _$rootScope_) ->
+      httpBackend = _$httpBackend_
+      GISService = _GISService_
+      $q = _$q_
+      $rootScope = _$rootScope_
+      return
+    )
+
+    describe 'getGISData', ->
+      describe 'when the attempt to get GIS data is successful', ->
+        afterEach ->
```

Comment:
@callaghanc Could this be moved to just below `describe 'getGISData'` so it applies to all tests for this method?

Reply:
@creativecoder yes you're right, i looked into the jasmine docs further and tested putting the `afterEach` on line 55 and removing all subsequent identical `afterEach`s and it does indeed run the block after every `it` within the line 54 `describe` regardless of nesting. i also learned that one can use `this` in `before/after` and subsequent `it`s to share state, which i didn't know before! was always putting general vars outside them for shared state. thanks for making me expand and correct my jasmine knowledge! 😊